usr.sbin/bsdinstall/scripts/cheri-desktop

```diff
@@ -0,0 +1,91 @@
+#!/bin/sh
+#
+# SPDX-License-Identifier: BSD-2-Clause
+#
+# Copyright (c) 2022 Jessica Clarke <jrtc27@FreeBSD.org>
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+# 1. Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+# 2. Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE AUTHOR AND CONTRIBUTORS ``AS IS'' AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED.  IN NO EVENT SHALL THE AUTHOR OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+# OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+# HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+# OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+# SUCH DAMAGE.
+#
+
+############################################################ INCLUDES
+
+BSDCFG_SHARE="/usr/share/bsdconfig"
+. $BSDCFG_SHARE/common.subr || exit 1
+f_dprintf "%s: loading_includes..." "$0"
+f_include $BSDCFG_SHARE/dialog.subr
+
+############################################################ MAIN
+
+: ${BSDDIALOG_OK=0}
+: ${BSDDIALOG_CANCEL=1}
+
+bsddialog --backtitle "$OSNAME Installer" --title "CHERI Desktop" --yesno \
+    "Would you like to install the CHERI desktop environment?" 0 0
+if [ $? -ne $BSDDIALOG_OK ]; then
+	exit 0
+fi
+
+# Running before config stage so make sure chroot has working DNS
+f_quietly cp -f $BSDINSTALL_TMPETC/resolv.conf $BSDINSTALL_CHROOT/etc/
+
+clear
+echo "$OSNAME Installer"
+echo "========================"
+echo "CHERI Desktop"
+echo
+ASSUME_ALWAYS_YES=YES chroot $BSDINSTALL_CHROOT pkg64c install cheri-desktop
```

Comment:
You can simplify this with:
```suggestion
chroot $BSDINSTALL_CHROOT pkg64c install -qy cheri-desktop
```

Reply:
1. I did ASSUME_ALWAYS_YES as that's what docsinstall does, along with a whole bunch of other parts of the system, since until 2018 the pkg wrapper wouldn't parse the -y during bootstrapping and ASSUME_ALWAYS_YES was the only way to make it not prompt
2. It takes a while to download and install the packages so once bootstrapped the user will get no output until it's done if -q is used, so they'll have no idea how long it's going to take and whether it's hung.

Reply:
pkg.cheribsd.org's performance is terrible (at least to the west coast of the US) so I think -q is not an option. We're talking 10 minutes or so to install because most packages download at less than 1MBps and even the largest never ramped as high as 3MBps (I routinely see 20MBps from other sources).